catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsMetacardSupport.java

```diff
@@ -13,17 +13,302 @@
+        for (ContentItem contentItem : incomingContentItems) {
+            try {
+                Path tmpPath = null;
+                long size;
+                try (InputStream inputStream = contentItem.getInputStream()) {
+                    if (inputStream == null) {
+                        throw new IngestException(
+                                "Could not copy bytes of content message.  Message was NULL.");
+                    }
+
+                    String sanitizedFilename =
+                            InputValidation.sanitizeFilename(contentItem.getFilename());
+                    tmpPath = Files.createTempFile(FilenameUtils.getBaseName(sanitizedFilename),
+                            FilenameUtils.getExtension(sanitizedFilename));
+                    Files.copy(inputStream, tmpPath, StandardCopyOption.REPLACE_EXISTING);
+                    size = Files.size(tmpPath);
+                    tmpContentPaths.put(contentItem.getId(), tmpPath);
+                } catch (IOException e) {
+                    if (tmpPath != null) {
+                        FileUtils.deleteQuietly(tmpPath.toFile());
+                    }
+                    throw new IngestException("Could not copy bytes of content message.", e);
+                }
+                String mimeTypeRaw = contentItem.getMimeTypeRawData();
+                mimeTypeRaw = guessMimeType(mimeTypeRaw, contentItem.getFilename(), tmpPath);
+
+                if (!InputValidation.checkForClientSideVulnerableMimeType(mimeTypeRaw)) {
+                    throw new IngestException("Unsupported mime type.");
+                }
+
+                String fileName = updateFileExtension(mimeTypeRaw, contentItem.getFilename());
+                Metacard metacard = generateMetacard(mimeTypeRaw,
+                        contentItem.getId(),
+                        fileName,
+                        subject,
+                        tmpPath);
+                metacardMap.put(metacard.getId(), metacard);
+
+                ContentItem generatedContentItem = new ContentItemImpl(metacard.getId(),
+                        com.google.common.io.Files.asByteSource(tmpPath.toFile()),
+                        mimeTypeRaw,
+                        fileName,
+                        size,
+                        metacard);
+                contentItems.add(generatedContentItem);
+            } catch (Exception e) {
+                tmpContentPaths.values()
+                        .forEach(path -> FileUtils.deleteQuietly(path.toFile()));
+                tmpContentPaths.clear();
+                throw new IngestException("Could not create metacard.", e);
+            }
+        }
+    }
+
+
+    /**
+     * Updates any empty metacard attributes with those defined in the
+     * {@link DefaultAttributeValueRegistry}.
+     *
+     * @param metacard the metacard to update with default attribute values
+     */
+    void setDefaultValues(Metacard metacard) {
```

Comment:
Maybe we should add this method to the `DefaultAttributeValueRegistry` interface and implementation instead of having it here?